crates/engine/src/resolver/graphql/deserialize/errors.rs

```diff
@@ -86,7 +64,16 @@ where
                 error
             })
             .collect();
-        self.0.response().set_subgraph_errors(errors);
+        self.response.borrow_mut().set_subgraph_errors(errors);
         Ok(errors_count)
     }
 }
+
+#[derive(serde::Deserialize)]
+struct SubgraphGraphqlError {
+    message: String,
+    #[serde(default)]
+    path: serde_json::Value,
```

Comment:
Also nitpick, but we should be able to assume `Vec<String>` for the path (https://spec.graphql.org/draft/#sec-Errors.Error-Result-Format)

Reply:
Kind of the same reason actually, we don't want deserialization to fail because of an invalid data format. This would stop the whole deserialization and not consume the rest of the JSON. So we need to accept anything and deal with problems as they come. So this actually needs to be also reworked (doing that currently for the data part). Thanks for pointing it out^^